src/components/NodeList.js

```diff
@@ -18,55 +19,100 @@ const EnhancedNode = DragSource(selectable(Node));
 /**
   * Renders a list of Node.
   */
-const NodeList = ({
-  nodes,
-  nodeColor,
-  label,
-  selected,
-  onSelect,
-  itemType,
-  isOver,
-  willAccept,
-  meta,
-}) => {
-  const isSource = !!find(nodes, ['uid', get(meta, 'uid', null)]);
-
-  const classNames = cx(
-    'node-list',
-    { 'node-list--hover': !isSource && willAccept && isOver },
-    { 'node-list--drag': !isSource && willAccept }, // TODO: rename class
-  );
-
-  return (
-    <StaggeredTransitionGroup
-      className={classNames}
-      component="div"
-      delay={animation.duration.fast * 0.2}
-      duration={animation.duration.slow}
-      start={animation.duration.slow + 3}
-      transitionName="node-list--transition"
-      transitionLeave={false}
-    >
-      { isOver && willAccept &&
-        <Node key="placeholder" placeholder />
-      }
-      {
-        nodes.map(node => (
-          <span key={node.uid}>
-            <EnhancedNode
-              color={nodeColor}
-              label={label(node)}
-              selected={selected(node)}
-              onSelected={() => onSelect(node)}
-              meta={() => ({ ...node, itemType })}
-              {...node}
-            />
-          </span>
-        ))
-      }
-    </StaggeredTransitionGroup>
-  );
-};
+class NodeList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      nodes: props.nodes,
+      refresh: 0,
+      stagger: true,
+    };
+
+    this.refreshTimer = null;
+  }
+
+  componentWillReceiveProps(newProps) {
+    // Don't update if nodes are the same
+    if (isEqual(map(newProps.nodes, 'uid'), map(this.state.nodes, 'uid'))) {
+      return;
+    }
+
+    // if we are on the same prompt/provided same id, then just append the node
```

Comment:
Is prompt id enough, or do we need stage and prompt?

Reply:
The NodeList id isn't required to reference prompt/stage. I've updated the comment to express that, but updated our usage to include stage. It isn't actually needed because components are not reused between stages, but I think it makes the intention clearer.